docs/reference.md

```diff
@@ -303,6 +311,71 @@ Specify the config file to write to. Without this option, your configuration
 will be written to `~/.gemstash/config.yml`. If you write to a custom location,
 you will need to pass the `--config-file` option to all Gemstash commands.
 
+## Preload
+
+Preload all the gems in your gemstash server from the default upstream.
+This can be useful if you plan to go somewhere where the internet connection
+will not be reliable or available.
+It is worth noting that this will require a large amount of disk space and
+computing time.
+
+### Usage
+
+```
+gemstash preload
+gemstash preload --latest
+gemstash preload --limit=1000
+gemstash preload --limit=1000 --skip=1000
+gemstash preload --threads=10 --server_url=http://my-server-url:9292
+```
+
+### Options
+
+#### --latest
+
+**Usage:** `--latest`
+
+**Description**<br />
+Only fetch the latest specs. This will limit the amount of disk space required,
+but there is no guarantee that the gems you are using will have all their
+dependencies met.
+
+#### --skip
+
+**Usage:** `--skip=<number>`
+
+**Description**<br />
+Number of gems to skip when preloading. This can be useful to load in chunks or
+to resume a gem preloading.
+
+#### --limit
+
+**Usage:** `--limit=<number>`
+
+**Description**<br />
+Limit for the number of gems to preload. In conjunction with [--skip](#--skip)
+this can be used for loading gems in chunks or resuming the process.
+
+#### --threads
+
+**Usage:** `--threads=<number>`
+
+**Description**<br />
+Number of threads to run the fetching job with.
+Since this activity is heavily IO bound (most of the time goes into waiting the
+network) it is useful to use more threads for performing the task. The default
+number of threads is 20 already which should be good enough, but in case you
+want to experiment this value can be changed.
+
+#### --server_url
```

Comment:
I think Thor converts underscores to dashes, so it would be `--server-url` (you can check by running `gemstash help preload`).